src/substrait/textplan/PlanPrinterVisitor.cpp

```diff
@@ -71,6 +71,19 @@ std::string visitEnumArgument(const std::string& str) {
   return text.str();
 }
 
+bool isAggregate(const SymbolInfo* symbol) {
+  if (symbol->subtype.type() == typeid(::substrait::proto::Rel::RelTypeCase) &&
+      ANY_CAST(::substrait::proto::Rel::RelTypeCase, symbol->subtype) ==
+          ::substrait::proto::Rel::kAggregate) {
+    return true;
+  }
```

Comment:
For pattern matching like this, I recommend adding a conditional cast macro:

```c++
#define ANY_CAST_IF(ValueType, value) \
  value.type() != typeid(ValueType) \
  ? ::std::nullopt \
  : ::std::make_optional(ANY_CAST(ValueType, value))
```

Then you can write
```suggestion
  if (auto type_case = ANY_CAST_IF(::substrait::proto::Rel::RelTypeCase, symbol->subtype)) {
    return type_case == ::substrait::proto::Rel::kAggregate;
  }
```

If you want to be even fancier, I can provide a `match()` helper:

```c++
bool isAggregate(const SymbolInfo* symbol) {
  return match(symbol->subtype,
    [](::substrait::proto::Rel::RelTypeCase type_case) {
      return type_case == ::substrait::proto::Rel::kAggregate;
    },
    [](RelationType rel_type) {
      return rel_type == RelationType::kAggregate;
    },
    [](...) {
      return false;
    });
}
```

Reply:
ANY_CAST_IF is roughly the amount of tech that I think I need at this point given how often I use ANY_CAST_IF (and I should be fixing things to use zero).  I am much appreciative of the ANY_CAST_IF structure -- definitely is more usable/readable.